writer/starlark.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package writer
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// StarlarkWriter is a simple type for writing basic Starlark macros with a consistent form.
+type StarlarkWriter struct {
+	w            *bufio.Writer
+	buf          []string
+	hasBody      bool
+	currentMacro string
+	dirStack     []string
+}
+
+// NewStarlarkWriter creates a new StarlarkWriter writing to the provided output.
+func NewStarlarkWriter(w io.Writer) *StarlarkWriter {
+	return &StarlarkWriter{bufio.NewWriter(w), nil, false, "", nil}
```

Comment:
you could replace with &StarlarkWriter{w: bufio.NewWriter(w)} if you don't feel the need to be explicit with the default values